### PR TITLE
python: add test showing that attaching a debugger works

### PR DESCRIPTION
**Pull request by tgummerer**
The debugging support has been merged in https://github.com/pulumi/pulumi/pull/17179, without adding tests to facilitate early testing.  We should however really have a test for it to make sure this works.  Add that here.

### CODE REVIEW

File: tests/integration/integration_python_test.go [hunk 1]
```diff
@@ -1460,3 +1468,137 @@ func TestConfigGetterOverloads(t *testing.T) {
...
+	conn, err := net.Dial("tcp", "localhost:"+strconv.Itoa(
+		int(debugEvent.Config["connect"].(map[string]interface{})["port"].(float64))))
+	if err != nil {
+		log.Fatalf("Failed to connect to debugger: %v", err)
+	}
+	defer conn.Close()
+
+	reader := bufio.NewReader(conn)
+	resp, err := dap.ReadProtocolMessage(reader)
+	assert.NoError(t, err)
```
Comment (lunaris) [inline tests/integration/integration_python_test.go:1521]: Nit but it'd be good to use only one of `require` vs `assert`

File: tests/integration/integration_python_test.go [hunk 2]
```diff
@@ -1460,3 +1468,137 @@ func TestConfigGetterOverloads(t *testing.T) {
 	// Run a preview. This will typecheck the program and fail if typechecking has errors.
 	e.RunCommand("pulumi", "preview")
 }
+
+func TestDebuggerAttachPython(t *testing.T) {
```
Comment (lunaris) [inline tests/integration/integration_python_test.go:1472]: Might be good to add a small doc comment explaining what we're testing here -- it looks like we just spin up a program in debug mode, then tell it to run and wait for it to terminate (using appropriate debug events) -- is that right?

**Review by lunaris — APPROVED** [review]

File: tests/integration/integration_python_test.go (hunk 2 above)
Comment (tgummerer) [reply tests/integration/integration_python_test.go:1472]: Yup, that's right.  It's to make sure the command runs in debug mode and responds to debug commands, and completes successfully after the debugger tells it to.  I'll add a comment.

File: tests/integration/integration_python_test.go (hunk 1 above)
Comment (tgummerer) [reply tests/integration/integration_python_test.go]: :heavy_check_mark:

**Pull request merged**